src/main/java/com/epam/aidial/core/Proxy.java

```diff
@@ -119,7 +119,7 @@ private void handleRequest(HttpServerRequest request) {
         } else if (apiKey != null) {
             key = config.getKeys().get(apiKey);
 
-            if (key == null) {
+            if (key == null && !config.existDeploymentApiKey(apiKey)) {
```

Comment:
We are changing essential authorization part. Would be nice to see at least some unit tests.

Reply:
fixed